lib/kube/proxy/forwarder.go

```diff
@@ -493,6 +507,45 @@ func (f *Forwarder) setupContext(ctx auth.Context, req *http.Request, isRemoteUs
 	return authCtx, nil
 }
 
+func (f *Forwarder) authorize(ctx context.Context, actx *authContext) error {
+	if actx.teleportCluster.isRemote {
+		// Authorization for a remote kube cluster will happen on the remote
+		// end (by their proxy), after that cluster has remapped used roles.
+		f.WithField("auth_context", actx.String()).Debug("Skipping authorization for a remote kubernetes cluster name")
+		return nil
+	}
+	if actx.kubeCluster == "" {
+		// This should only happen for remote clusters (filtered above), but
+		// check and report anyway.
+		f.WithField("auth_context", actx.String()).Debug("Skipping authorization due to unknown kubernetes cluster name")
+		return nil
+	}
+	servers, err := f.AccessPoint.GetKubeServices(ctx)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	// Check authz against the first match.
+	//
+	// We assume that users won't register two identically-named clusters with
```

Comment:
What would be the best place to verify that? I think if a user falls into this trap, there should be a way to tell.